sycl-fusion/common/include/Kernel.h

```diff
@@ -20,6 +20,28 @@ namespace jit_compiler {
 
 using BinaryAddress = const uint8_t *;
 
+/// Possible barrier flags
+enum class BarrierFlags : int {
+  None = 0,
+  Local = 1,
+  Global = 2,
+  LocalAndGlobal = Local | Global
+};
+
+constexpr BarrierFlags getNoBarrierFlag() { return BarrierFlags::None; }
+constexpr BarrierFlags getLocalAndGlobalBarrierFlag() {
+  return BarrierFlags::LocalAndGlobal;
+}
+constexpr bool isNoBarrierFlag(BarrierFlags Flag) {
+  return Flag == BarrierFlags::None;
+}
+constexpr bool hasLocalBarrierFlag(BarrierFlags Flag) {
+  return static_cast<int>(Flag) & static_cast<int>(BarrierFlags::Local);
+}
+constexpr bool hasGlobalBarrierFlag(BarrierFlags Flag) {
+  return static_cast<int>(Flag) & static_cast<int>(BarrierFlags::Global);
+}
+
```

Comment:
Can we use `uint32_t` instead of `int` for underlying type for consistence? (also makes more sense for bitwise arithetic)

Reply:
Type is consistent with the enum class's type. But changed to `uint32_t` anyway.